### PR TITLE
Add web compat unit tests and fixes

### PR DESCRIPTION
**Pull request by necolas**
## Summary

Adds compat with W3C logical CSS properties. See https://github.com/facebook/react-native/issues/34425

This is a replacement for reverted #34590, which can no longer be imported internally.

## Changelog

[General][Added] - Added CSS logical properties.

## Test Plan

Unit test snapshots.

### CODE REVIEW

File: Libraries/Components/View/View.js
Comment (necolas) [inline Libraries/Components/View/View.js:14]: rename to `processStyles`

File: Libraries/Components/View/View.js
Comment (necolas) [inline Libraries/Components/View/View.js:103]: rename to `processStyles`

File: Libraries/Image/Image.android.js
Comment (necolas) [inline Libraries/Image/Image.android.js:16]: rename to `processStyles`

File: Libraries/Image/Image.android.js
Comment (necolas) [inline Libraries/Image/Image.android.js:170]: rename to `processStyles`

File: Libraries/Image/Image.ios.js
Comment (necolas) [inline Libraries/Image/Image.ios.js:17]: rename to `processStyles`

File: Libraries/Image/Image.ios.js
Comment (necolas) [inline Libraries/Image/Image.ios.js:141]: rename to `processStyles`

File: Libraries/StyleSheet/__tests__/processStyles-test.js
Comment (necolas) [inline Libraries/StyleSheet/__tests__/processStyles-test.js:13]: rename to `processStyles`

File: Libraries/Components/TextInput/TextInput.js
Comment (NickGerleman) [inline Libraries/Components/TextInput/TextInput.js:22]: Could we choose a different prefix than process? processColor, processTransform, etc have a [specific meaning](https://github.com/facebook/react-native/blob/9f3a3e13cc7e083dc0151b6037fc483dbfeaeabc/Libraries/Components/View/ReactNativeStyleAttributes.js), where they are used by the view config to transform a value before being passed to native.

`processStyles` instead does some JS side transformation, and it is not clear without reading the code where it should be used, or what it does from the name.

File: Libraries/Components/TextInput/TextInput.js
Comment (NickGerleman) [inline Libraries/Components/TextInput/TextInput.js:1405]: nit: we should chose a different way than underscore to avoid shadowing. The ESLint rules special case variables starting with underscore as allowed to be possibly unused, and it is commonly used in the codebase for instance-like variables.

File: Libraries/Components/TextInput/TextInput.js
Comment (NickGerleman) [inline Libraries/Components/TextInput/TextInput.js:1435]: If `processStyles` requires a style to always be flattened, maybe it should do the flattening itself? Though see the other comments on naming/function.

File: Libraries/StyleSheet/StyleSheetTypes.js
Comment (NickGerleman) [inline Libraries/StyleSheet/StyleSheetTypes.js:276]: nit: we can link to the 1:1 prop. E.g. https://developer.mozilla.org/en-US/docs/Web/CSS/padding-block-end

File: Libraries/Components/TextInput/TextInput.js
Comment (NickGerleman) [reply Libraries/Components/TextInput/TextInput.js:22]: React Native also allows custom native components which inherit layout related processing from the native side. So these mapping will not work correctly or be exposed to primitives outside the package.

Do we have any concrete plan to eventually bring these to native? Even without the shorthand in Yoga, doing this mapping on the layer of `LayoutableShadowNode` on the native side would make this all work correctly on Fabric without needing to translate each component.

**Review by NickGerleman — COMMENTED** [review]
Left some feedback on implementation (e.g. `processStyles` should probably be named something different) but more generally think for the case of adding layout shorthands, it would make sense to skip the JS shimming and just do the change in native.

Mapping at the current layer means that these properties will not work outside the fixed set added here, and makes the complexity of mapping properties a cross-cutting to existing internal components. Apart from external components being able to use the mapping functions, custom native components can inherit support for Yoga styling properties on the native side without ever touching explicit styles on the JS layer.

A more correct solution to this that would still be relatively straightforward would be to forward these styles to the native component, then doing the mapping to supported Yoga properties in Fabric's C++ layer. That single implementation would work on all platforms, but not all of OSS (or even some of Meta's apps) are on Fabric yet, so a native solution supporting paper would need two more implementations on the Java and ObjC side.

Depending on how forward-looking we would like to be, this could still being useful for staging, butI don't think this change leaves a great terminal state compared to doing the mapping in `YogaLayoutableShadowNode` instead. Longer term, these should probably all be in Yoga, but its current structure makes adding even shortand props increase total app memory usage.

**Comment by necolas** [comment]
This diff was already merged and I've resubmit with a fix and tests. I agree that in the long term all the props and styles should be done in native, but in the meanwhile they've all been done in JS. I don't think we should block this patch after already merging 30+ props and styles also shimmed in JS

**Comment by NickGerleman** [comment]
I know it was merged before, but I would still like to review the change 😉.

There are negative properties of JS shimming as a solution for layout properties specifically. It adds newly inconsistent behavior for native components (you can flexbox layout arbitrary native views), along with some debt in our components. It also means we need to rebuild a solution once Yoga supports the shorthand.

None of this is the end of the world, but this seems to be building a foundation for more changes. If we are adding more of these, we can avoid a lot of debt, rework, and correctness issues if we instead implement these layout properties as a couple of switch cases in native (and I am willing to add those).

File: Libraries/Components/TextInput/TextInput.js
Comment (necolas) [reply Libraries/Components/TextInput/TextInput.js:22]: > Could we choose a different prefix than process?

Open to suggestions

**Comment by necolas** [comment]
@NickGerleman This diff contains tests needed to meet an internal SLA. What do you want to be changed in order to get this merged ASAP? Shall I remove the layout prop shim in favor of the draft PR you posted?

**Comment by NickGerleman** [comment]
> @NickGerleman This diff contains tests needed to meet an internal SLA. What do you want to be changed in order to get this merged ASAP? Shall I remove the layout prop shim in favor of the draft PR you posted?

Yeah, I'd prefer we hold off on merging the layout reprojection portion until a couple more details get ironed out. But I would sign off the rest, being the tests and fixes for existing implemented props.

**Review by NickGerleman — APPROVED** [review]

**Pull request closed without merging**
